tests/containers/workbenches/workbench_image_test.py

```diff
@@ -31,21 +32,48 @@ class TestWorkbenchImage:
         # disable ipv6 https://danwalsh.livejournal.com/47118.html
         {"net.ipv6.conf.all.disable_ipv6": "1"}
     ])
-    def test_image_entrypoint_starts(self, image: str, sysctls) -> None:
+    def test_image_entrypoint_starts(self, subtests: pytest_subtests.SubTests, image: str, sysctls) -> None:
         skip_if_not_workbench_image(image)
 
+        # Here is a list of blocked keywords we don't want to see in the log messages during the container/workbench
+        # startup (e.g. log messages from Jupyter IDE, code-server IDE or RStudio IDE).
+        blocked_keywords = ["Error", "error", "Warning", "warning", "Failed", "failed", "[W ", "[E ", "Traceback"]
+        # Here is a list of allowed messages that match some block keyword from the list above, but we allow them
+        # for some reason.
+        allowed_messages = [
+            # We don't touch this - it should be fixed in the future by the package upgrade.
+            "JupyterEventsVersionWarning: The `version` property of an event schema must be a string. It has been type coerced, but in a future version of this library, it will fail to validate."
+        ]
+
         container = WorkbenchContainer(image=image, user=1000, group_add=[0],
                                        sysctls=sysctls)
         try:
             try:
                 container.start()
                 # check explicitly that we can connect to the ide running in the workbench
-                container._connect()
+                with subtests.test("Attempting to connect to the workbench..."):
+                    container._connect()
             finally:
                 # try to grab logs regardless of whether container started or not
                 stdout, stderr = container.get_logs()
+                full_logs = ""
                 for line in stdout.splitlines() + stderr.splitlines():
-                    logging.debug(line)
+                    full_logs = "\n".join([full_logs, line.decode()])
+                    logging.debug(line.decode())
+
+                # let's check that logs don't contain any error or unexpected warning message
+                with subtests.test("Checking the log in the workbench..."):
+                    failed_lines = 0
```

Comment:
on second thought, maybe I'd make this

```suggestion
                    failed_lines = []
```

and .append to it the failed lines

why should people hunt for them in the log when they can be kept here and produced in the final failure message

Reply:
yup, I like it - updated